generator/src/test/java/com/scottlogic/deg/generator/cucumber/formattedAs.feature

```diff
@@ -18,11 +18,11 @@ Feature: User can specify that a value is so formatted
 #      | 1.0                      | "%a"         | "0x1.0p0"                      | no way to specify float over double
 #      | 1.5                      | "%a"         | "0x1.8p0"                      | no way to specify float over double
       | null                     | "%a"         | null                           |
-      | 1                        | "%b"         | true                           |
-      | "1"                      | "%b"         | true                           |
-      | 2018-10-10T00:00:00.000  | "%b"         | true                           |
-#      | null                     | "%b"         | false                           | this should be false
-      |                          | "%b"         | true                           |
+      | 1                        | "%b"         | "true"                         |
```

Comment:
Hi, what are the cucumber tests supposed to cover? I have concerns around changing the format of the expected values. For example if we were generating JSON, I would expect to have true not surrounded by "" but perhaps this is covered by another type of testing. Similarly with the difference between "no value" () and "empty string" (""). The other thing I noticed is that the place where we have made the change to allow bools/disallow + integers is in parseInput in GeneratorTestUtilities but this is not used in all the places where we parse values (it is only used to parse set values, invalid fields and expected values) - for example if I use +12 or true with equals to operator, I just get "this step is undefined"  rather than "exception". This seems to automatically Ignore the test rather than fail it. A similar behaviour is observed when + prefixed integers are used with less than constraint (for example) too.

Scenario: Running an 'equalTo' request that includes roman character strings that include in-use values ("true") should be successful
     Given there is a field foo
       And foo is equal to "true"
     Then the following data should be generated:
       | foo    |
       | "true" |

cucumber.runtime.junit.UndefinedThrowable: The step "foo is equal to true" is undefined



Scenario: Running an 'equalTo' request that includes roman character strings that include in-use values ("true") should be successful
     Given there is a field foo
       And foo is equal to +12
     Then the following data should be generated:
       | foo    |
       | 12 |

cucumber.runtime.junit.UndefinedThrowable: The step "foo is equal to +12" is undefined

Scenario: User requires to create a numeric field with data values that are greater than zero
     Given there is a field foo
       And foo is greater than 0
       And foo is less than +10
       And foo is granular to 1
       And foo is anything but null
     Then the following data should be included in what is generated:
       | foo |
       | 1   |
       | 2   |
       | 3   |
       | 4   |
       | 5   |
       | 6   |
       | 7   |
       | 8   |
       | 9   |


cucumber.runtime.junit.UndefinedThrowable: The step "foo is less than +10" is undefined

Reply:
I would expect this message to appear given our current configuration, as the + symbol before a number is not supported.

> The step "foo is less than +10" is undefined

I've had a brief look at ways of making the Cucumber framework treat ignored steps as failures. They are currently reported as failures, but not at the step level - i.e. the scenario will be reported as a failure only. The 'undefined' step and following steps (defined or not) will be reported as ignored.

Reply:
In comment to your earlier question about true/false. The following should be supported, if not then there is an issue, please provide the scenario so i can debug.

* true --> TRUE - converted to boolean
* false --> FALSE - converted to boolean
* "true" --> "true" - left as is (string)
* "false" --> "false" - left as is (string)